reference/cd-github.md

```diff
@@ -0,0 +1,92 @@
+---
+title: GitHub Workflow (Preview)
+---
+
+<style>
+    .img-bordered {
+        border: 2px solid rgba(0,0,0,0.15);
+    }
+    .move-right {
+        margin-left: 32px;
+    }
+    .move-right-small {
+        margin-left: 8px;
+    }
+</style>
+
+Pulumi's GitHub workflow integration is a feature which shows the results of Pulumi stack
+updates on GitHub. Showing you any potential infrastructure changes on Pull Requests,
+as well as recording stack changes made with each push.
+
+> GitHub workflow integration is currently in preview, and we will be expanding the feature set
+> over the comming weeks and months. If you have any feedback, suggestions, or bug reports, please
+> let us know in the [Pulumi Community Slack](https://slack.pulumi.io/).
+
+
+**Pull Request Comment**
+
+![Comment on Pull Request](/images/github-app/pr-comment.png){:class="img-bordered"}{:class="move-right"}
+
+**Check Suite Result**
+
+![Results on GitHub Check](/images/github-app/checks-detail.png){:width="600px"}{:class="img-bordered"}{:class="move-right"}
+
+## Installation and Configuration
+
+Pulumi's GitHub workflow integration is a GitHub application you can install by visiting
+[github.com/apps/pulumi](https://github.com/apps/pulumi) or clicking the button below.
+
+<a href="https://github.com/apps/pulumi" target="_blank">
+    <button class="button">INSTALL</button>
+</a>
+
+The Pulumi GitHub application is installed into a specific GitHub organization, and you can
+configure it to only be used by certain repositories.
+
+![Installation Page](/images/github-app/installation.png){:width="450px"}{:class="img-bordered"}{:class="move-right"}
+![Configuration Page](/images/github-app/org-configuration.png){:width="450px"}{:class="img-bordered"}{:class="move-right-small"}
+
+The Pulumi GitHub application does not have access to your source code. And will only report
+status on pushes / pull requests that happen for repositories it is configured to access.
+
+You can also uninstall the GitHub application at any time without impacting your stacks or
+other Pulumi-managed resources.
+
+## CI Integration
+
+Once the Pulumi GitHub application is installed in your organization, any `pulumi preview` or
+`pulumi update` that is ran in your CI system will have its results reported back to GitHub.
+
+Currently, Pulumi's GitHub application only supports Travis CI. (For instructions on configuring
```

Comment:
This was surprising to me. Why does this require Travis?

Reply:
It has to do with how we filter results to just those ran in CI. e.g. we don't want `pulumi preview` results you run on your desktop from showing up on GitHub.

We do that by only triggering the workflow results that have CI-specific metadata config values, which we only set for Travis. But it's super easy to support other CI systems using this model.

https://github.com/pulumi/pulumi/blob/9d84f2e2490c2f3f86985b2f35393856b01d52b7/cmd/util.go#L437